t/local_package.t

```diff
@@ -67,44 +68,66 @@ subtest "constructor tests" => sub {
 };
 
 subtest 'refresh and unpack index files' => sub {
+    plan skip_all => "IO::Uncompress::Gunzip is not available"
+      unless $CPAN::Common::Index::Mirror::HAS_IO_UNCOMPRESS_GUNZIP;
     my $index = new_local_index;
 
-    ok( !-e File::Spec->catfile( $cache, $packages ), "$packages not in cache" );
+    ok( !-e File::Spec->catfile( $index->cache, $packages ), "$packages not in cache" );
```

Comment:
This is a subtle semantic change in the test here and on lines 79, 85, and 90.  Originally, the test is making sure that the file is/isn't in the directory we passed to the constructor.  The new test could completely ignore the cache attribute in the constructor as long as the cache *method* returns a directory where things actually do/don't live.  So we know longer have a assurance that the cache constructor attribute is respected.

You could address that in `new_local_index` and `new_uncompressed_local_index` by testing that `$index->cache eq $cache` before returning `$index`.